lib/Semantics/check-select-stmt.cpp

```diff
@@ -0,0 +1,393 @@
+//===-- lib/semantics/check-select-stmt.cc --------------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include "check-select-stmt.h"
+#include "flang/Parser/message.h"
+#include "flang/Semantics/tools.h"
+
+namespace Fortran::semantics {
+
+class SelectCaseHelper {
+
+public:
+  SelectCaseHelper(parser::Messages &messages, int kind, TypeCategory category)
+    : kind_{kind}, category_(category), messages_{messages} {}
+
+  template<typename T> std::string GetValueAsString(T val) {
+    if constexpr (std::is_same_v<T, bool>) {
+      return std::string(val ? ".TRUE." : ".FALSE.");
+    } else if constexpr (std::is_same_v<T, std::int64_t>) {
+      return std::to_string(val);
+    } else {
+      return parser::QuoteCharacterLiteral(val);
+    }
+  }
+  template<typename T> std::string GetValuePairAsString(T fromVal, T toVal) {
+    if constexpr (std::is_same_v<T, std::optional<std::int64_t>>) {
+      return std::string((fromVal ? std::to_string(*fromVal) : "") + ":" +
+          (toVal ? std::to_string(*toVal) : ""));
+    } else {
+      return ((fromVal ? parser::QuoteCharacterLiteral(fromVal.value())
+                       : std::string("")) +
+          std::string(":") +
```

Comment:
`":"s` would be better, here and elsewhere.

Reply:
I will replace all occurrences.